scxml_converter/src/scxml_converter/scxml_entries/scxml_data.py

```diff
@@ -141,14 +141,20 @@ def check_validity(self) -> bool:
         if get_data_type_from_string(self._data_type) is None:
             print(f"Error: SCXML data: '{self._id}' has unknown type '{self._data_type}'.")
             return False
-        valid_expr = is_non_empty_string(ScxmlData, "expr", self._expr)
+        if isinstance(self._expr, str):
+            valid_expr = is_non_empty_string(ScxmlData, "expr", self._expr)
+        else:
+            valid_expr = isinstance(self._expr, (int, float))
+            if not valid_expr:
+                print(f"Error: SCXML data: '{self._id}' ",
+                      "default type is invalid: '{type(self._expr)}'.")
```

Comment:
Why `default type is invalid`? What is the `default type`?

Reply:
The error message is indeed confusing, I tried to make it clearer.
What I was trying to say there is that the initial expression of a variable must be either a string or an int / float value (when initialized from code), and the error should report when the type is neither of the above.
Additionally, I guess we should allow bool as well, but adding that to the ValidTypes would create issues with the bounds, so for now it is not considered

Reply:
Sounds good. Yes booleans are quite important (I am not sure which use cases have them, but I would expect they all do). Does this mean that at the moment any boolean variable in the datamodel breaks the compilation?

Reply:
Since I noticed the flaw with bools, I adapted ScxmlData to support bools as well, now it should be clearer